### PR TITLE
Add logging for Consul 5XX retries

### PR DESCRIPTION
**Pull request by Ashald**
As simple as subj says. Bounced the idea with @adleong in Slack before creating this PR.

### CODE REVIEW

File: consul/src/main/scala/io/buoyant/consul/v1/BaseApi.scala [hunk 1]
```diff
@@ -26,11 +26,17 @@ trait BaseApi extends Closable {
   private[this] val infiniteRetryFilter = new RetryFilter[http.Request, http.Response](
     RetryPolicy.backoff(backoffs) {
       // We will assume 5xx are retryable, everything else is not for now
-      case (_, Return(rep)) => rep.status.code >= 500 && rep.status.code < 600
+      case (req, Return(rep)) if rep.status.code >= 500 && rep.status.code < 600 =>
+        log.error(
+          s"Retrying Consul request '${req.method} ${req.uri}' on '${rep.status.code}' response"
```
Comment (adleong) [inline consul/src/main/scala/io/buoyant/consul/v1/BaseApi.scala]: I'm torn on if we should include the `rep.contentString` as well here.  `ConsulApiError` does so maybe we should include that here as well.

Comment (Ashald) [inline consul/src/main/scala/io/buoyant/consul/v1/BaseApi.scala]: Yeah, that's great idea!

Comment (Ashald) [inline consul/src/main/scala/io/buoyant/consul/v1/BaseApi.scala]: How do you like it now?

Comment (adleong) [inline consul/src/main/scala/io/buoyant/consul/v1/BaseApi.scala]: 👌

**Pull request merged**